src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +112,11 @@ public int Count
             }
         }
 
+        public ref T this[int index] => ref _array[_offset + index];
```

Comment:
Note that this indexer will have poor performance characteristics on reference types for *both* reading and writing because of array covariance (it will have array covariance check for both). @KrzysztofCwalina Have you considered this during the API review?

I think this should be the classic get/set; forget about ref indexer here.